### ISSUE TITLE
Ignore umlauts and diacritics

### ISSUE THREAD

**Issue by skjerns** [cboxdoerfer/fsearch#467]
Sometimes, files have Umlauts (üöä) in them or contain diacritics (éáó) which might be difficult to type on an US/ISO keyboard. It would be great if search could be performed on the base chars, i.e. when searching for `resumee`, `resumée.pdf` should also appear. Even better would be that in the reverse case when searching for `resuméé`, `resumee.pdf` does not pop up.

Do you think some functionality like this could be implemented?

**Comment by cboxdoerfer** [comment]
Yes, that's definitely something I'd like to add. But this a bit of work, because internally FSearch uses four different code paths to compare strings:

* the default substring matching for ASCII characters: For that to support ignoring diacritics we need to add a custom `strstr` (and `strcasestr`) implementation, where we have full control over how diacritics are handled. That shouldn't be any problem or affect performance too much.
* unicode substring matching: I'm pretty sure the ICU library I'm using for that supports ignoring diacritics already, but this likely comes at a performance cost. 
* regex matching: FSearch is using the PCRE library for that and AFAIK it doesn't support ignoring of diacritics and it's not really feasible to implement a custom regex engine. But I think the diacritics can be stripped from strings before they get passed to the PCRE library, however this comes at a performance cost.
* wildcard matching: this actually uses the PCRE library as well (wildcard patterns are simply converted to regex patterns), so there's no additional work for that case necessary

As I'm focusing on the 0.3 release with file system monitoring at the moment, I won't be able to work on it soon. But once 0.3 is out I'll have a look at that.
